core/src/main/scala/com/acervera/osm4scala/DenseNodesIterator.scala

```diff
@@ -26,78 +26,115 @@
+import com.acervera.osm4scala.utilities.DecompressUtils.{decompressChangeset, decompressCoord, decompressTimestamp, decompressUid, decompressUserSid, iteratorCheck}
+import org.openstreetmap.osmosis.osmbinary.osmformat.{DenseInfo, DenseNodes, StringTable}
 
 /**
   * Iterator over a DenseDataNode block.
-  * By default, lanOffset, longOffset and graularity is 0, 0 and 100 because I did not found pbf files with other values.
   *
   * @param osmosisStringTable
   * @param osmosisDenseNode
   * @param latOffset
   * @param lonOffset
   * @param granularity
+  * @param dateGranularity
   */
-class DenseNodesIterator(osmosisStringTable: StringTable,
+case class DenseNodesIterator(osmosisStringTable: StringTable,
                          osmosisDenseNode: DenseNodes,
-                         latOffset: Long = 0,
-                         lonOffset: Long = 0,
-                         granularity: Int = 100)
+                         latOffset: Option[Long] = None,
+                         lonOffset: Option[Long] = None,
+                         granularity: Option[Int] = None,
+                         dateGranularity: Option[Int] = None)
     extends Iterator[NodeEntity] {
 
-  if (osmosisDenseNode.denseinfo.isDefined && osmosisDenseNode.denseinfo.get.visible.nonEmpty) {
-    throw new Exception("Only visible nodes are implemented.")
-  }
-
-  private val idIterator = osmosisDenseNode.id.iterator
-  private val lonIterator = osmosisDenseNode.lon.iterator
-  private val latIterator = osmosisDenseNode.lat.iterator
-  private val tagsIterator = osmosisDenseNode.keysVals.iterator
-
-  private var lastNode: NodeEntity = NodeEntity(0, 0, 0, Map())
+  private val idIterator: Iterator[Long] = osmosisDenseNode.id.iterator
+  private val lonIterator: Iterator[Long]  = osmosisDenseNode.lon.iterator
+  private val latIterator: Iterator[Long]  = osmosisDenseNode.lat.iterator
+  private val tagsIterator: Iterator[Int] = osmosisDenseNode.keysVals.iterator
+  private val optionDenseInfo: Option[DenseInfo] = osmosisDenseNode.denseinfo
+  private val versionIterator: Iterator[Int] = if(optionDenseInfo.isDefined) optionDenseInfo.get.version.iterator else Iterator.empty
+  private val timestampIterator: Iterator[Long] = if(optionDenseInfo.isDefined) optionDenseInfo.get.timestamp.iterator else Iterator.empty
+  private val changesetIterator: Iterator[Long]  = if(optionDenseInfo.isDefined) optionDenseInfo.get.changeset.iterator else Iterator.empty
+  private val uidIterator: Iterator[Int]  = if(optionDenseInfo.isDefined) optionDenseInfo.get.uid.iterator else Iterator.empty
+  private val userSidIterator: Iterator[Int]  = if(optionDenseInfo.isDefined) optionDenseInfo.get.userSid.iterator else Iterator.empty
+  private val visibleIterator: Iterator[Boolean]  = if(optionDenseInfo.isDefined) optionDenseInfo.get.visible.iterator else Iterator.empty
+
+  private val _latOffSet: Long = latOffset.getOrElse[Long](DenseNodesIterator.DEFAULT_LAT_OFFSET)
```

Comment:
If you think that the right name of your variable is the same than an existing one, it is a smell code.
`_` is a _usually_ bad practice in Scala. In other languages like Python or Dart is used for scoping but Scala has proper keywords to do it. So if `_` does not add semantic to the name (usually does not, like in this case), is better to think about why and refactor your code when finding the reason of the smell.

Reply:
That make sense even I also see a lot of usage like this in scale code. I will try to concise it